specification/communication/data-plane/Chat/preview/2025-11-15-preview/communicationserviceschat.json

```diff
@@ -0,0 +1,2210 @@
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Post typing event to a thread": {
+            "$ref": "./examples/Messages_SendTypingNotification.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/images": {
+      "post": {
+        "tags": [
+          "Images"
+        ],
+        "summary": "Upload an image in a thread, on behalf of a user.",
+        "operationId": "ChatThread_UploadChatImage",
+        "consumes": [
+          "application/octet-stream"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "header",
+            "name": "image-filename",
+            "description": "The file name of the image.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "chatThreadId",
+            "description": "Thread id where the uploaded image belongs to. (Teams meeting only)",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "chatImageFile",
+            "description": "Image binary data, allowed image formats: jpeg, png, gif, heic, webp",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ChatImageFile"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Uploaded successfully, the `Location` header contains the URL for the newly uploaded image.",
+            "schema": {
+              "$ref": "#/definitions/UploadChatImageResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Upload Image": {
+            "$ref": "./examples/Images_UploadChatImage.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/images/{imageId}/view/{imageViewType}": {
+      "get": {
+        "tags": [
+          "Images"
+        ],
+        "summary": "Get an image by view type.",
+        "operationId": "ChatThread_GetChatImage",
+        "produces": [
+          "application/json",
+          "image/gif",
+          "image/jpeg",
+          "image/png",
+          "image/bmp",
+          "image/tiff"
```

Comment:
Are these content-type deduced from filename in the `upload` API above?
Could it be wrong (the type here mismatch from the binary)?

Reply:
This list are the supported image formats that GetImage operation is able to return. The above upload API also checks for supported image format validates the file before the upload can proceed, type mismatch wouldn't happen in the get API as it returns the image binary data that was stored in the media service.